src/NServiceBus.NHibernate/Subscriptions/CachedSubscriptionPersister.cs

```diff
@@ -19,16 +19,14 @@ public CachedSubscriptionPersister(ISessionFactory sessionFactory, TimeSpan expi
 
         public override Task Subscribe(Subscriber subscriber, MessageType messageType, ContextBag context)
         {
-            base.Subscribe(subscriber, messageType, context);
             cache.Clear();
-            return Task.FromResult(0);
+            return base.Subscribe(subscriber, messageType, context);
```

Comment:
Is the sequence relevant here? For example, previously a failed subscribe (exception) would not clear the cache and now it does. Is this intentional?